sandbox/libs/authn/src/main/java/org/opensearch/authn/internal/InternalSubject.java

```diff
@@ -63,10 +64,29 @@ public String toString() {
     /**
      * Logs the user in via authenticating the user against current Shiro realm
      */
-    public void login(AuthenticationToken authenticationToken) {
+    public void login(AuthenticationToken authenticationToken) throws RuntimeException {
         org.apache.shiro.authc.AuthenticationToken authToken = AuthenticationTokenHandler.extractShiroAuthToken(authenticationToken);
         // Login via shiro realm.
-        SecurityUtils.getSecurityManager().authenticate(authToken);
-        // shiroSubject.login(authToken);
+        ensureUserIsLoggedOut();
```

Comment:
There is no need for `ensureUserIsLoggedOut` because in the `shiro.ini` file it is set to be stateless using:

```
securityManager.subjectDAO.sessionStorageEvaluator.sessionStorageEnabled = false
```